### PR TITLE
Fix collector_id overflow panic

### PR DESCRIPTION
**Pull request by breeswish**
See https://github.com/pingcap/rust-prometheus/issues/236

### CODE REVIEW

File: src/registry.rs [hunk 1]
```diff
@@ -93,11 +93,11 @@ impl RegistryCore {
 
     fn unregister(&mut self, c: Box<Collector>) -> Result<()> {
         let mut id_set = Vec::new();
-        let mut collector_id = 0;
+        let mut collector_id: u64 = 0;
         for desc in c.desc() {
             if id_set.iter().find(|id| **id == desc.id).is_none() {
                 id_set.push(desc.id);
-                collector_id += desc.id;
+                collector_id = collector_id.wrapping_add(desc.id);
```
Comment (nrc) [inline src/registry.rs]: I think it is better to use `checked_add` and return an Error, rather than silently wrapping around.

Comment (breeswish) [reply src/registry.rs]: It is intended to overflow

**Comment by breeswish** [comment]
@overvenus @siddontang PTAL

**Comment by breeswish** [comment]
ping @overvenus @siddontang

**Comment by nrc** [comment]
@breeswish can we merge this PR and release 0.6 now?

**Pull request merged**